abn/service_impl.go

```diff
@@ -0,0 +1,206 @@
+func splitApplicationKey(applicationKey string) (string, string) {
+	var name, namespace string
+	names := strings.Split(applicationKey, "/")
+	if len(names) > 1 {
+		namespace, name = names[0], names[1]
+	} else {
+		namespace, name = "default", names[0]
+	}
+
+	return namespace, name
+}
+
+//
+//
+//
+
+// writeMetricInternal is detailed implementation of gRPC method WriteMetric
+func writeMetricInternal(application, user, metric, valueStr string) error {
+	log.Logger.Tracef("writeMetricInternal called for application, user: %s, %s", application, user)
+	defer log.Logger.Trace("writeMetricInternal completed")
+
+	s, track, err := lookupInternal(application, user)
+	if err != nil || track == nil {
+		log.Logger.Warnf("lookupInternal failed for application=%s, user=%s", application, user)
+		return err
+	}
+	log.Logger.Debugf("lookupInternal(%s,%s) -> %d", application, user, *track)
+
+	value, err := strconv.ParseFloat(valueStr, 64)
+	if err != nil {
+		log.Logger.Warn("Unable to parse metric value ", valueStr)
+		return err
+	}
+
+	v := s.Versions[*track]
+	transaction := uuid.NewString()
+
+	err = metricsClient.SetMetric(
+		s.Namespace+"/"+s.Name, *track, *v.Signature,
+		metric, user, transaction,
+		value)
+
+	if err != nil {
+		log.Logger.Warnf("Unable to set metric %s for application=%s, user=%s", metric, application, metric)
+	}
+
+	return nil
+}
+
+func toLegacyApplication(s *controllers.Routemap) *abnapp.LegacyApplication {
+	name := s.Namespace + "/" + s.Name
+	tracks := make(abnapp.LegacyTracks, len(s.Versions))
+	versions := make(abnapp.LegacyVersions, len(s.Versions))
+	for t, v := range s.Versions {
+		asStr := fmt.Sprintf("%d", t)
+		tracks[asStr] = asStr
+
+		vms, err := metricsClient.GetSummaryMetrics(name, t, *v.Signature)
+		if err != nil {
+			return nil
+		}
+		metrics := make(map[string]*summarymetrics.SummaryMetric, len(vms.MetricSummaries))
+		for metric, summary := range vms.MetricSummaries {
+			metrics[metric] = &summarymetrics.SummaryMetric{
+				float64(summary.SummaryOverTransactions.Count),
+				float64(summary.SummaryOverTransactions.Count) * summary.SummaryOverTransactions.Mean,
+				summary.SummaryOverTransactions.Min,
+				summary.SummaryOverTransactions.Max,
+				summary.SummaryOverTransactions.StdDev, // should be sum of squares
+			}
+		}
+
+		versions[asStr] = &abnapp.LegacyVersion{
+			Metrics: metrics,
+		}
+	}
+
+	a := &abnapp.LegacyApplication{
+		Name:     name,
+		Tracks:   tracks,
+		Versions: versions,
+	}
+	return a
+}
+
+// getApplicationDataInternal is detailed implementation of gRPC method GetApplicationData
+func getApplicationDataInternal(application string) (string, error) {
+
+	namespace, name := splitApplicationKey(application)
+	s := controllers.AllRoutemaps.GetRoutemapFromNamespaceName(namespace, name)
+	if s == nil {
+		return "", fmt.Errorf("routemap not found for application %s", namespace+"/"+name)
+	}
+
+	legacyApp := toLegacyApplication(s)
```

Comment:
Since the task is going away, we don't need to worry about the legacyApplication at all. Everything to do with the A/B/n SDK will be through the metrics DB, and Grafana (not through the experiment).

Reply:
this has been removed